xwiki-platform-core/xwiki-platform-test/pom.xml

```diff
@@ -42,7 +42,14 @@
     <module>xwiki-platform-test-integration</module>
     <module>xwiki-platform-test-oldcore</module>
     <module>xwiki-platform-test-page</module>
-    <module>xwiki-platform-test-ui</module>
-    <module>xwiki-platform-test-upgrade</module>
   </modules>
+  <profiles>
+    <profile>
+      <id>integration-tests</id>
+      <modules>
+        <module>xwiki-platform-test-ui</module>
+        <module>xwiki-platform-test-upgrade</module>
```

Comment:
If we do this, we should also move test-docker in the `integration-tests` profile IMO.

Reply:
The suggested change will prevent building these modules when not using the integration tests profiles, so a bit more build perf (always good to have though :)).

Now the root problem is not this one. The root problem is that the upgrade test should not depend on something in extension IMO. It could be fixed in another issue though.

Reply:
I also moved platform-test-docker. I let you create the other issue.